File: submissions/time_limit_exceeded/tle.py
groundelms=set(input().split())
combinations=int(input())
revrecipies={}
for _ in range(combinations):
    # fire water steam
    a, b, c = input().split()
    revrecipies[c] = tuple(sorted([a, b]))
def sol(rec: dict[str, tuple[str, str]], curr: str):
    if curr in groundelms:
        return 1
    else:
        a, b = rec[curr]
        return sol(rec, a) + sol(rec, b)

elm=input()
print(sol(revrecipies, elm))

File: submissions/time_limit_exceeded/tle.cpp
#include <iostream>
#include <set>
#include <map>
#include <string>

using namespace std;

int sol(map<string, pair<string, string>> &rec, string curr) {
    if (rec.find(curr) == rec.end()) {
        return 1;
    } else {
        string a = rec[curr].first;
        string b = rec[curr].second;
        return sol(rec, a) + sol(rec, b);
    }
}

int main() {
    int amountgroundelms;
    cin >> amountgroundelms;
    set<string> groundelms;
    for (int i = 0; i < amountgroundelms; i++) {
        string elm;
        cin >> elm;
        groundelms.insert(elm);
    }

    int combinations;
    cin >> combinations;
    map<string, pair<string, string>> revrecipies;
    for (int i = 0; i < combinations; i++) {
        string a, b, c;
        cin >> a >> b >> c;
        revrecipies[c] = make_pair(min(a, b), max(a, b));
    }

    string elm;
    cin >> elm;
    cout << sol(revrecipies, elm) - 1 << endl;

    return 0;
}
